source/features/enable-file-links-in-compare-view.tsx

```diff
@@ -0,0 +1,120 @@
+import React from 'dom-chef';
+import select from 'select-dom';
+import delegate from 'delegate-it';
+import * as pageDetect from 'github-url-detection';
+
+import features from '.';
+import GitHubURL from '../github-helpers/github-url';
+
+function isPRMenuOpening(event: delegate.Event): void {
+	const dropdown = event.delegateTarget.nextElementSibling!;
+
+	// Only if it's not already there
+	if (select.exists('.rgh-actionable-link', dropdown)) {
+		return;
+	}
+
+	// Only enabled on Open/Draft PRs. Editing files doesn't make sense after a PR is closed/merged.
+	if (!select.exists('.gh-header-meta [title$="Open"], .gh-header-meta [title$="Draft"]')) {
+		return;
+	}
+
+	// If you're viewing changes from partial commits, ensure you're on the latest one.
+	const isPartialCommits = select.exists('.js-commits-filtered');
+	if (isPartialCommits && !select.exists('[aria-label="You are viewing the latest commit"]')) {
+		return;
+	}
+
+	// This solution accounts for:
+	// - Branches with slashes in it
+	// - PRs opened from the default branch
+	const viewFile = select<HTMLAnchorElement>('[data-ga-click^="View file"]', dropdown)!;
+	const url = new GitHubURL(viewFile.href);
+	const [user, repository, ...branchParts] = select<HTMLAnchorElement>('.commit-ref.head-ref a')!.title.replace(':', '/').split('/');
+	const branch = branchParts.join('/');
+	url.assign({
+		user,
+		repository,
+		branch
+	});
+
+	viewFile.href = String(url);
+
+	viewFile.classList.add('rgh-actionable-link'); // Mark this as processed
+}
+
+function isCompareMenuOpening(event: delegate.Event): void {
+	const dropdown = event.delegateTarget.nextElementSibling!;
+
+	// Only if it's not already there
+	if (select.exists('.rgh-actionable-link', dropdown)) {
+		return;
+	}
+
+	const viewFile = select<HTMLAnchorElement>('[data-ga-click^="View file"]', dropdown)!;
+	const url = new GitHubURL(viewFile.href);
+	url.assign({
+		branch: location.pathname.replace(/.*:/, '')
+	});
+	viewFile.href = String(url);
+	viewFile.classList.add('rgh-actionable-link'); // Mark this as processed
+
+	// If the allow maintainers to edit is not available your either comparing a Permalink and the file can't be edited or your comparing a fork you can't edit the file on
```

Comment:
Reword or fix errors when you have a chance

Reply:
Ok I will try to explain.
When I am doing here is making the edit and delete buttons work. However if you are comparing a Permalink you cant edit the file on that nor delete it. 
Another case where you cant edit the file would be if you dont have right to edit the repo.

I looked through the dom and I saw that when creating a PR the `allow maintainers to edit` is not there in those cases.
However my solution will fail when creating a PR on your own repo.

Reply:
In the code, not as a comment.